Makefile

```diff
@@ -259,6 +259,15 @@ generate-mocks: mockery ## Generate the mocks for the project as configured in .
 clean-ui:
 	find ${UI_DIR} -type f -name extension.tar -delete
 
+PHONY: codegen-download-openapi
+codegen-download-openapi: run-backend
+	@echo "Downloading OpenAPI spec from dev server"
+	yarn --cwd ${UI_DIR}  api:download
+
+.PHONY: codegen-ui
+codegen-ui: ## Generate the UI API files.
```

Comment:
it seems that this target is dependant on the `build-ui` one as I was getting the following error:
```
❯ make codegen-ui                                                                                   
yarn --cwd /Users/lalmeida1/dev/git/intuit/argocd-ephemeral-access/ui  api:generate
yarn run v1.22.22
warning package.json: No license field
$ orval
/bin/sh: orval: command not found
error Command failed with exit code 127.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [codegen-ui] Error 127
```

 If so, please add the `build-ui` as a dependency for this target

Reply:
After running the following commands:
```
make build-ui
make codegen-ui
```
I am getting the following error:
```
❯ make codegen-ui                                                                                              
yarn --cwd /Users/lalmeida1/dev/git/intuit/argocd-ephemeral-access/ui  api:generate
yarn run v1.22.22
warning package.json: No license field
$ orval
🍻 Start orval v7.1.1 - A swagger client generator for typescript
File not found => orval.config.{js,mjs,ts}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [codegen-ui] Error 1
```

It seems that this `orval.config.{js,mjs,ts}` file needs to exist somewhere

Reply:
add codegen-ui as prerequisite for build-ui